src/net.cpp

```diff
@@ -2865,3 +2867,25 @@ uint64_t CConnman::CalculateKeyedNetGroup(const CAddress& ad) const
 
     return GetDeterministicRandomizer(RANDOMIZER_ID_NETGROUP).Write(vchNetGroup.data(), vchNetGroup.size()).Finalize();
 }
+
+void LogMessage(CNode* node, std::string& command, std::vector<unsigned char>& data, bool is_incoming)
+{
+    int64_t time = GetTimeMicros();
+
+    fs::path base_path = GetDataDir() / "peer_logging" / node->addr.ToString();
+    boost::filesystem::create_directories(base_path);
```

Comment:
use `fs::create_directories` (which is currently just an alias for boost::filesystem`, but means that the std filesystem can be substituted in when we upgrade our c++ version.

Reply:
Nit. If if log messages where enabled  by` -logmessages=<LogMessageDir> `not empty
we would have a separate LogMessagesDir , instead of logging in DataDir by default. And DataDir should be always <> LogDir.imho. see   #19419

Reply:
> 
> 
> Nit. If if log messages where enabled by`-logmessages=<LogMessageDir>`not empty
> we would have a separate LogMessagesDir , instead of logging in DataDir by default. And DataDir should be always <> LogDir.imho. see #19419

I'll consider this, having the log directory configurable would be nice.